sql-cli/noxfile.py

```diff
@@ -21,15 +21,45 @@ def dev(session: nox.Session) -> None:
 
 
 @nox.session(python=["3.7", "3.8", "3.9"])
-def test(session: nox.Session) -> None:
+@nox.parametrize("airflow", ["2.2.5", "2.3.4", "2.4.2"])
+def test(session: nox.Session, airflow: str) -> None:
     """Run both unit and integration tests."""
+
     session.install("poetry")
-    session.run("poetry", "install", "--with", "dev")
-    # Log all the installed dependencies
+    if airflow.startswith("2.2"):
+        # To install some versions of Airflow, we need constraints, due to issues like:
+        # https://github.com/apache/airflow/issues/19804
+        constraints_url = (
+            "https://raw.githubusercontent.com/apache/airflow/"
+            f"constraints-{airflow}/constraints-{session.python}.txt"
+        )
+
+        # Poetry does not support constraints:
+        # https://github.com/python-poetry/poetry/issues/3225
+        session.install("-e", ".", f"apache-airflow=={airflow}", "-c", constraints_url)
+
+        # We are duplicating the tests dependencies until we find a better solution.
+        # The solution might be to move out of poetry.
```

Comment:
I tried to automate this, and I didn't manage - since this ticket has been long-standing, I would really like us to just merge it and focus on higher priority ones.